elastic/kibana
# [Security solution] Cases webhook, include additional fields

## Summary

Resolves https://github.com/elastic/kibana/issues/154054

Cases webhook connector users [have requested](https://discuss.elastic.co/t/sharing-case-id-value-using-elastic-case-management-webhook/328641/3) other case fields, such as the case id, should be exposed as options in the different request JSON body inputs. Currently we only expose title, description, and tags. This PR adds `id`, `severity`, and `status` to be available as well. 


### To test:

1. Use [these credentials](https://p.elstc.co/paste/rjHkw13x#WayiUPhHJg6JSkfDhZC2ZOWU886pivvA-7MQo3gZD6P) to create a Cases Webhook connector to Jira. Notice the new fields (`id`, `severity`, and `status`) are included in the Create and Update JSON, mapped to Jira custom fields. 
2. Use the following credentials to login to Jira so we can check the results:  https://p.elstc.co/paste/Z3TPI88c#5sp7k7ZOmTpi7+UIC9nucXrI8ZK2hvwWlnxSJU9Cklv
3. Back in Kibana, create a case and push to the Jira Webhook connector you created. Open the case in Jira, ensure the Kibana Status, Kibana Severity, and Kibana Ids are filled with the correct value. This is located in the right sidebar:
    <img width="1289" alt="Screenshot 2023-09-13 at 10 22 12 AM" src="https://github.com/elastic/kibana/assets/6935300/7f5764f4-6ee5-4962-b97e-66c6b666c24e">
4.  One at a time, update `status` and `severity` and ensure the "Requires update to Jira Webhook incident" message appears in user actions:
    <img width="863" alt="Screenshot 2023-09-12 at 1 53 50 PM" src="https://github.com/elastic/kibana/assets/6935300/a3d8a587-44cb-45df-96ea-b6a03b9f43e8">
5. Press the "Update Jira Webhook incident" button. Ensure the updates are made in the Jira incident
6. Create a standard (not cases webhook) Jira connector, assign it to the case you've been testing with (so now the case has 2 connectors). Push to the standard Jira connector
7. Update the case severity. Ensure that the "Requires update" message appears for Jira Webhook but not the standard Jira, ex:
    <img width="1214" alt="Screenshot 2023-09-13 at 9 58 41 AM" src="https://github.com/elastic/kibana/assets/6935300/d179fc85-b5c5-4a6d-9ae7-4d50b4405353">
